File: uis/client/web2/src/utils/textCaseUtils.ts
// src/utils/textCaseUtils.ts
// Utility functions for text case management

/**
 * Converts a string to uppercase.
 */
export function toUpperCase(str: string): string {
  return str.toUpperCase();
}

/**
 * Converts a string to lowercase.
 */
export function toLowerCase(str: string): string {
  return str.toLowerCase();
}

/**
 * Converts a string to title case (first letter of each word capitalized).
 */
export function toTitleCase(str: string): string {
  return str.replace(/\w\S*/g, (txt) => txt.charAt(0).toUpperCase() + txt.substr(1).toLowerCase());
}

/**
 * Converts a string to camelCase.
 */
export function toCamelCase(str: string): string {
  return str
    .replace(/[-_\s]+(.)?/g, (_, c) => (c ? c.toUpperCase() : ''))
    .replace(/^(.)/, (m) => m.toLowerCase());
}

/**
 * Converts a string to snake_case.
 */
export function toSnakeCase(str: string): string {
  return str
    .replace(/([a-z])([A-Z])/g, '$1_$2')
    .replace(/\s+/g, '_')
    .toLowerCase();
}

/**
 * Converts a string to kebab-case.
 */
export function toKebabCase(str: string): string {
  return str
    .replace(/([a-z])([A-Z])/g, '$1-$2')
    .replace(/\s+/g, '-')
    .toLowerCase();
}

File: uis/client/mobile_app/lib/utils/text_case_utils.dart
// Export functions for easy import
export 'text_case_utils.dart';
// lib/utils/text_case_utils.dart
// Utility functions for text case management in Dart

/// Converts a string to uppercase.
String toUpperCase(String str) => str.toUpperCase();

/// Converts a string to lowercase.
String toLowerCase(String str) => str.toLowerCase();

/// Converts a string to title case (first letter of each word capitalized).
String toTitleCase(String str) {
  return str.replaceAllMapped(
    RegExp(r'\w\S*'),
    (Match match) => match.group(0)![0].toUpperCase() + match.group(0)!.substring(1).toLowerCase(),
  );
}

/// Converts a string to camelCase.
String toCamelCase(String str) {
  final words = str.split(RegExp(r'[-_\s]+'));
  if (words.isEmpty) return '';
  return words.first.toLowerCase() +
      words.skip(1).map((w) => w.isEmpty ? '' : w[0].toUpperCase() + w.substring(1).toLowerCase()).join();
}

/// Converts a string to snake_case.
String toSnakeCase(String str) {
  return str
      .replaceAllMapped(RegExp(r'([a-z])([A-Z])'), (m) => '${m[1]}_${m[2]}')
      .replaceAll(RegExp(r'\s+'), '_')
      .toLowerCase();
}

/// Converts a string to kebab-case.
String toKebabCase(String str) {
  return str
      .replaceAllMapped(RegExp(r'([a-z])([A-Z])'), (m) => '${m[1]}-${m[2]}')
      .replaceAll(RegExp(r'\s+'), '-')
      .toLowerCase();
}
